### PR TITLE
Fixed images in black margins

### LINKED ISSUE #27440: Image instead of black bars - VisualServer
**Godot version:**
3.1 stable

**OS/device including version:**
Windows 10 

**Issue description:**
I am having problem setting texture to be used instead of black bars when application window is scaled. It always draws one color of texture(instead of the texture itself) like if it was zoomed in or something. Is this normal behaviour or a bug? I tried textures with defferent resolutions and also images instead of textures but nothing seems to work. 

Example - godot icon is used as texture for black bars
godot_window

Normally i would expect that texture that is passed to visualServer will be scaled to the size corresponding bar instead of this.

**Steps to reproduce:**
```
var txt = preload("res://icon.png")

func _ready():
	VisualServer.black_bars_set_images( txt .get_rid(),  txt .get_rid(),  txt .get_rid(),  txt .get_rid())
	VisualServer.black_bars_set_margins(50,50,50,50)
```

### PR DESCRIPTION
**Pull request by azagaya**
Fixes #27440. Also, i could add the option so stretch image to margin size, but would require new methods to set that option.

### CODE REVIEW

**Comment by azagaya** [comment]
Oops... i added a white space at the beginning. Let me fix it.

**Review by clayjohn — APPROVED** [review]
Looks good to me.

**Comment by akien-mga** [comment]
Could you squash the two commits into one?

**Comment by azagaya** [comment]
Oh, sorry... I think it should be fine now.

File: drivers/gles2/rasterizer_canvas_base_gles2.cpp [hunk 1]
```diff
@@ -250,7 +250,8 @@ void RasterizerCanvasBaseGLES2::draw_window_margins(int *black_margin, RID *blac
 	if (black_image[MARGIN_LEFT].is_valid()) {
 		_bind_canvas_texture(black_image[MARGIN_LEFT], RID());
 		Size2 sz(storage->texture_get_width(black_image[MARGIN_LEFT]), storage->texture_get_height(black_image[MARGIN_LEFT]));
-		draw_generic_textured_rect(Rect2(0, 0, black_margin[MARGIN_LEFT], window_h), Rect2(0, 0, sz.x, sz.y));
+		draw_generic_textured_rect(Rect2(0, 0, black_margin[MARGIN_LEFT], window_h),
+				Rect2(0, 0, (float)black_margin[MARGIN_LEFT] / sz.x, (float)(window_h) / sz.y));
```
Comment (akien-mga) [inline drivers/gles2/rasterizer_canvas_base_gles2.cpp:254]: Are we confident that `sz.x` and `sz.y` can't ever be 0?

Comment (akien-mga) [reply drivers/gles2/rasterizer_canvas_base_gles2.cpp:254]: I tested with this and it did not crash nor return any error, so I guess there's some validation that happens further up the stack:
```gdscript
	var rid = VisualServer.texture_create()
	print(rid.get_id())  # 147
	print(VisualServer.texture_get_height(rid))  # 0
	print(VisualServer.texture_get_width(rid))  # 0
	
	VisualServer.black_bars_set_images(rid, rid, rid, rid)
	VisualServer.black_bars_set_margins(50,50,50,50)
```

**Comment by akien-mga** [comment]
Tested locally and it does fix the bug, and works fairly well. I've noticed some weird artifacts when resizing but this might be due to Linux/KWin issues. I'd be interested in a test project as mentioned in #27440 to see if I can confirm that issue and open a bug report.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**